pkg/flare/archive_dca.go

```diff
@@ -176,3 +187,78 @@ func getDCAWorkloadList() ([]byte, error) {
+func getAgentDaemonSet(fb flarehelpers.FlareBuilder) error {
+	// The Agent DaemonSet name is based on the Helm chart template and added to the Cluster Agent as an environment variable
+	var agentDaemonsetName string
+	var releaseNamespace string
+	var agentDaemonset []byte
+
+	cl, err := apiserver.GetAPIClient()
+	if err != nil {
+		log.Info("Can't create client to query the API Server: %s", err)
+	} else {
+		agentDaemonsetName = os.Getenv(HELM_AGENT_DAEMONSET)
+		releaseNamespace = os.Getenv(HELM_CHART_RELEASE_NAMESPACE)
+		agentDaemonset, err = apiserver.GetDs(cl, agentDaemonsetName, releaseNamespace)
+		if err != nil {
+			log.Debugf("Error while collecting the Agent DaemonSet: %q", err)
+		}
+	}
+
+	return fb.AddFile("agent-daemonset.yaml", agentDaemonset)
+}
+
+// getClusterAgentDeployment retrieves the Deployment manifest of the Cluster Agent
+func getClusterAgentDeployment(fb flarehelpers.FlareBuilder) error {
+	// The Cluster Agent Deployment name is based on the Helm chart template and added to the Cluster Agent as an environment variable
+	var clusterAgentDeploymentName string
+	var releaseNamespace string
+	var clusterAgentDeployment []byte
+
+	cl, err := apiserver.GetAPIClient()
+	if err != nil {
+		log.Info("Can't create client to query the API Server: %s", err)
+	} else {
+		clusterAgentDeploymentName = os.Getenv(HELM_CLUSTER_AGENT_DEPLOYMENT)
+		releaseNamespace = os.Getenv(HELM_CHART_RELEASE_NAMESPACE)
+		clusterAgentDeployment, err = apiserver.GetDeploy(cl, clusterAgentDeploymentName, releaseNamespace)
+		if err != nil {
+			log.Debugf("Error while collecting the Cluster Agent Deployment: %q", err)
+		}
+	}
+
+	return fb.AddFile("cluster-agent-deployment.yaml", clusterAgentDeployment)
+}
+
+// getHelmValues retrieves the user-defined values for the Datadog Helm chart
+func getHelmValues(fb flarehelpers.FlareBuilder) error {
+	var helmUserValues []byte
+	var releaseName string
+	var releaseNamespace string
+
+	cl, err := apiserver.GetAPIClient()
+	if err != nil {
+		log.Info("Can't create client to query the API Server: %s", err)
+	} else {
+		releaseName = os.Getenv(HELM_CHART_RELEASE_NAME)
+		releaseNamespace = os.Getenv(HELM_CHART_RELEASE_NAMESPACE)
+		helmUserValues, err = apiserver.GetDatadogHelmUserValuesFromSecret(cl, releaseName, releaseNamespace)
+		// The returned bytes array is not nil, we can exit
+		if len(helmUserValues) != 0 {
+			return fb.AddFile("helm-values.yaml", helmUserValues)
+		}
+		// The cluster Agent was unable to retrieve Helm chart data from secrets, attempting to retrieve them from Configmaps
+		helmUserValues, err = apiserver.GetDatadogHelmUserValuesFromConfigmap(cl, releaseName, releaseNamespace)
+		// The returned bytes array is not nil, we can exit
+		if len(helmUserValues) != 0 {
+			return fb.AddFile("helm-values.yaml", helmUserValues)
+		}
+		if err != nil {
+			log.Debugf("Error while collecting the Helm chart values: %q", err)
+		}
```

Comment:
IMO the `err` check should be before the `addFile`

```suggestion
		helmUserValues, err = apiserver.GetDatadogHelmUserValuesFromConfigmap(cl, releaseName, releaseNamespace)
		if err != nil {
			log.Warningf("Error while collecting the Helm chart values: %v", err)
		}
		// The returned bytes array is not nil, we can exit
		if len(helmUserValues) != 0 {
			return fb.AddFile("helm-values.yaml", helmUserValues)
		}
```